### PR TITLE
Return err when create ContainerLogsDir failed

### PR DESCRIPTION
**Pull request by fisherxu**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespace from that line:
>
> /kind api-change

/kind bug
> /kind cleanup
> /kind deprecation
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs
NONE
```

### CODE REVIEW

**Comment by fisherxu** [comment]
cc @yujuhong @dashpole @sjenning

**Comment by dashpole** [comment]
This change makes the error fatal for the kubelet.  Can you explain why we want to make that change?

File: pkg/kubelet/kubelet.go [hunk 1]
```diff
@@ -1269,7 +1269,7 @@ func (kl *Kubelet) initializeModules() error {
 	// If the container logs directory does not exist, create it.
 	if _, err := os.Stat(ContainerLogsDir); err != nil {
 		if err := kl.os.MkdirAll(ContainerLogsDir, 0755); err != nil {
-			klog.Errorf("Failed to create directory %q: %v", ContainerLogsDir, err)
+			return fmt.Errorf("Failed to create directory %q: %v", ContainerLogsDir, err)
```
Comment (SergeyKanzhelev) [inline pkg/kubelet/kubelet.go:1272]: also [this](https://github.com/golang/go/wiki/CodeReviewComments#error-strings):

```suggestion
			return fmt.Errorf("failed to create directory %q: %v", ContainerLogsDir, err)
```

**Comment by fisherxu** [comment]
@dashpole @SergeyKanzhelev Thanks for review :) If create `ContainerLogsDir` failed, kubelet will output the err log below. So we should make sure the `ContainerLogsDir` created correctly. In addition, all the errs in the func are returned, except this one. :) 
image

File: pkg/kubelet/kubelet.go (hunk 1 above)
Comment (fisherxu) [reply pkg/kubelet/kubelet.go]: Done.

**Comment by dashpole** [comment]
I looked through the history (this was added in #7123) to see if this was done for a particular reason.  It doesn't look like we chose to log vs return the error intentionally, and I agree with the author that we shouldn't start unless we can create required directories.

/lgtm
/approve

**Pull request merged**